aws/resource_aws_msk_cluster_test.go

```diff
@@ -185,6 +185,37 @@ func TestAccAWSMskCluster_ClientAuthentication_Tls_CertificateAuthorityArns(t *t
 	})
 }
 
+func TestAccAWSMskCluster_ClientAuthentication_Sasl_Scram(t *testing.T) {
+	var cluster1 kafka.ClusterInfo
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_msk_cluster.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSMsk(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMskClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMskClusterConfigClientAuthenticationSaslScram(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMskClusterExists(resourceName, &cluster1),
+					resource.TestCheckResourceAttr(resourceName, "client_authentication.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "client_authentication.0.sasl.0.scram", "1"),
```

Comment:
To correct the current test failure: 
```suggestion
					resource.TestCheckResourceAttr(resourceName, "client_authentication.0.sasl.0.scram", "true"),
```

Reply:
current failure I'm seeing for reference:
```
    TestAccAWSMskCluster_ClientAuthentication_Sasl_Scram: resource_aws_msk_cluster_test.go:193: Step 1/2 error: Check failed: Check 3/3 error: aws_msk_cluster.test: Attribute 'client_authentication.0.sasl.0.scram' expected "1", got "true"
--- FAIL: TestAccAWSMskCluster_ClientAuthentication_Sasl_Scram (1555.60s)
```